lib/src/rules/unnecessary_await_in_return.dart

```diff
@@ -96,7 +96,7 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
     if (returnType != null &&
         returnType.isDartAsyncFuture &&
-        type.isAssignableTo(returnType)) {
+        type.isSubtypeOf(returnType)) {
```

Comment:
Looks good to me but I'll defer to @bwilkerson for the final call since he may be aware of other implications.

Reply:
It seems reasonable to me, and the tests appear to show that it's the right test, but I haven't spent any time to try to _prove_ to myself that it's correct. I say, go for it.